### PR TITLE
init form properties and project tree

### PR DESCRIPTION
**Pull request by janicka1**
Feature #13827

Arvados-DCO-1.1-Signed-off-by: Janicki Artur <artur.janicki@contractors.roche.com>

### CODE REVIEW

File: src/views-components/search-bar/search-bar-advanced-view.tsx [hunk 1]
```diff
@@ -78,12 +79,13 @@ export const SearchBarAdvancedView = compose(
     reduxForm<SearchBarAdvanceFormData, SearchBarAdvancedViewActionProps>({
         form: SEARCH_BAR_ADVANCE_FORM_NAME,
         onSubmit: (data: SearchBarAdvanceFormData, dispatch: Dispatch) => {
+            console.log('data: ', data);
```
Comment (kowalczykpawel7) [inline src/views-components/search-bar/search-bar-advanced-view.tsx]: console.log

File: src/index.tsx [hunk 2]
```diff
@@ -114,6 +115,7 @@ const initListener = (history: History, store: RootStore, services: ServiceRepos
             initWebSocket(config, services.authService, store);
             await store.dispatch(loadWorkbench());
             addRouteChangeHandlers(history, store);
+            store.dispatch(initAdvanceFormProjectsTree());
```
Comment (burzomir) [inline src/index.tsx]: It probably should be placed in actions responsible for opening search box.

File: src/models/search-bar.ts [hunk 3]
```diff
@@ -4,15 +4,21 @@
...
     inTrash: boolean;
     dateFrom: string;
     dateTo: string;
     saveQuery: boolean;
     searchQuery: string;
+    properties: PropertyValues[];
+} & PropertyValues;
+
+export interface PropertyValues {
+    propertyKey: string;
```
Comment (burzomir) [inline src/models/search-bar.ts]: Wouldn't {key: string, value: string} be simpler?

File: src/store/search-bar/search-bar-actions.ts [hunk 4]
```diff
@@ -94,3 +97,18 @@ const getFilters = (filterName: string, searchValue: string): string => {
         .addEqual('groupClass', GroupClass.PROJECT, GroupContentsResourcePrefix.PROJECT)
         .getFilters();
 };
+
+export const initAdvanceFormProjectsTree = () => 
+    async (dispatch: Dispatch, getState: () => RootState, services: ServiceRepository) => {
```
Comment (burzomir) [inline src/store/search-bar/search-bar-actions.ts]: Why async? You're not using await inside this function.

File: src/store/search-bar/search-bar-actions.ts [hunk 5]
```diff
@@ -94,3 +97,18 @@ const getFilters = (filterName: string, searchValue: string): string => {
...
+        dispatch<any>(initUserProject(SEARCH_BAR_ADVANCE_FORM_PICKER_ID));
+    };
+
+export const changeAdvanceFormProperty = (property: string, value: PropertyValues[] | string = '') => 
+    async (dispatch: Dispatch, getState: () => RootState, services: ServiceRepository) => {
```
Comment (burzomir) [inline src/store/search-bar/search-bar-actions.ts]: Why async? You're not using await inside this function.

File: src/store/search-bar/search-bar-actions.ts [hunk 6]
```diff
@@ -94,3 +97,18 @@ const getFilters = (filterName: string, searchValue: string): string => {
...
+        dispatch(change(SEARCH_BAR_ADVANCE_FORM_NAME, property, value));
+    };
+
+export const updateAdvanceFormProperties = (propertyValues: PropertyValues) => 
+    async (dispatch: Dispatch, getState: () => RootState, services: ServiceRepository) => {
```
Comment (burzomir) [inline src/store/search-bar/search-bar-actions.ts]: Why async? You're not using await inside this function.

File: src/views-components/form-fields/search-bar-form-fields.tsx [hunk 7]
```diff
@@ -53,13 +62,13 @@ export const SearchBarDataToField = () =>
 
 export const SearchBarKeyField = () => 
     <Field
-        name='key'
+        name='propertyKey'
```
Comment (burzomir) [inline src/views-components/form-fields/search-bar-form-fields.tsx]: IMO `key` and `value` names were simpler.

File: src/views-components/search-bar/search-bar-advanced-properties-view.tsx [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+} from '~/store/search-bar/search-bar-actions';
+import { PropertyValues } from '~/models/search-bar';
+import { ArvadosTheme } from '~/common/custom-theme';
+import { SearchBarKeyField, SearchBarValueField } from '~/views-components/form-fields/search-bar-form-fields';
+import { Chips } from '~/components/chips/chips';
+
+type CssRules = 'root' | 'label' | 'button';
+
+const styles: StyleRulesCallback<CssRules> = (theme: ArvadosTheme) => ({
+    root: {
```
Comment (burzomir) [inline src/views-components/search-bar/search-bar-advanced-properties-view.tsx]: Empty object.

**Pull request merged**